daffodil-io/src/main/scala/org/apache/daffodil/io/InputSource.scala

```diff
@@ -325,8 +361,10 @@ class BucketingInputSource(inputStream: java.io.InputStream, bucketSize: Int = 1
     } else {
       val (bucketIndex, byteIndex) = bytePositionToIndicies(curBytePosition0b)
 
-      if (buckets(bucketIndex.toInt) == null)
-        throw new BacktrackingException
+      if ((bucketIndex < 0) || (buckets(bucketIndex.toInt) == null)) {
+        _isValid = false
```

Comment:
Should this _isValid change happen somewhere more generically instead? I'm thinking maybe in doParse(), where if the final parse result fails because of an SDE then the InputSource becomes invalid? While it's true that this specific case makes the InputSource invalid, any SDE (which this will eventually become one) also make it invalid.

Reply:
Missed this comment somehow.  Should be addressed in the most recent commit.